sky/check.py

```diff
@@ -78,14 +81,45 @@ def check_one_cloud(cloud_tuple: Tuple[str, clouds.Cloud]) -> None:
                        f'{enabled_clouds_str}[/green]')
 
 
+def get_enabled_clouds(
+        raise_if_no_cloud_access: bool = False) -> List[clouds.Cloud]:
+    """Returns a list of enabled clouds.
+
+    This function will perform a refresh if no public cloud is enabled.
+
+    Args:
+        raise_if_no_cloud_access: if True, raise an exception if no public
+            cloud is enabled.
+
+    Raises:
+        exceptions.NoCloudAccessError: if no public cloud is enabled and
+            raise_if_no_cloud_access is set to True.
+    """
+    if not global_user_state.get_cached_enabled_clouds():
+        try:
+            check(quiet=True)
+        except SystemExit:
+            # If no cloud is enabled, check() will raise SystemExit.
+            # Here we catch it and raise the exception later only if
+            # raise_if_no_cloud_access is set to True.
+            pass
+    cached_enabled_clouds = global_user_state.get_cached_enabled_clouds()
```

Comment:
nit: one less read in normal case.
```suggestion
    cached_enabled_clouds = global_user_state.get_cached_enabled_clouds()
    if not cached_enabled_clouds:
        try:
            check(quiet=True)
        except SystemExit:
            # If no cloud is enabled, check() will raise SystemExit.
            # Here we catch it and raise the exception later only if
            # raise_if_no_cloud_access is set to True.
            pass
        cached_enabled_clouds = global_user_state.get_cached_enabled_clouds()
```